ros2bag/ros2bag/verb/record.py

```diff
@@ -13,11 +13,29 @@
 # limitations under the License.
 
 import datetime
+import logging
 import os
+from typing import Dict
+from typing import Optional
 
+import rclpy
+from rclpy.duration import Duration
+from rclpy.qos import InvalidQoSProfileException
+from rclpy.qos import QoSProfile
 from ros2bag.verb import VerbExtension
-
 from ros2cli.node import NODE_NAME_PREFIX
+import yaml
+
+
+POLICY_MAP = {
+    'history': rclpy.qos.QoSHistoryPolicy.get_from_short_key,
+    'reliability': rclpy.qos.QoSReliabilityPolicy.get_from_short_key,
+    'durability': rclpy.qos.QoSDurabilityPolicy.get_from_short_key,
+    'liveliness': rclpy.qos.QoSLivelinessPolicy.get_from_short_key
+}
```

Comment:
How do we make sure those two will stay in-sync? Can't we just get the user to pass the short key?

Reply:
We can have the user specify in the overrid yaml file the enum integer for each policy (ex. `reliability: 0` which is the same as `reliability: system_default`) but that's not good UX.
I don't think there's a 'better' way to get these values beside using these helper methods provided by each policy.
Maybe using [`importlib`](https://docs.python.org/3/library/importlib.html) to import all modules with the name `*Policy` in `rclpy.qos` but that's hacky IMO.

Reply:
I see, we should at least add a comment explaining this needs to be updated when rclpy.qos is updated.
Consider also using a `frozenset()` here as you don't really need a map to store `x => x.foo` for multiple values of `x`, unless you plan to have elements that would break this pattern, but it's probably YAGNI.